builtin/providers/azurerm/resource_arm_container_service.go

```
@@ -0,0 +1,684 @@
+	}
+	for _, ssh := range *profile.SSH.PublicKeys {
+		keys := map[string]interface{}{}
+		keys["key_data"] = ssh.KeyData
+		sshKeys.Add(keys)
+	}
+
+	values["admin_username"] = profile.AdminUsername
+	values["ssh_key"] = sshKeys.List()
+	profiles.Add(values)
+
+	return &profiles
+}
+
+func flattenAzureRmContainerServiceAgentPoolProfiles(profiles *[]containerservice.AgentPoolProfile) *schema.Set {
+	agentPoolProfiles := schema.Set{
+		F: resourceAzureRMContainerServiceAgentPoolProfilesHash,
+	}
+
+	for _, profile := range *profiles {
+		agentPoolProfile := map[string]interface{}{}
+		agentPoolProfile["count"] = profile.Count
+		agentPoolProfile["dns_prefix"] = profile.DNSPrefix
+		agentPoolProfile["fqdn"] = profile.Fqdn
+		agentPoolProfile["name"] = profile.Name
+		agentPoolProfile["vm_size"] = string(profile.VMSize)
+		agentPoolProfiles.Add(agentPoolProfile)
+	}
+
+	return &agentPoolProfiles
+}
+
+func flattenAzureRmContainerServiceServicePrincipalProfile(profile *containerservice.ServicePrincipalProfile) *schema.Set {
+
+	if profile == nil {
+		return nil
+	}
+
+	servicePrincipalProfiles := schema.Set{
+		F: resourceAzureRMContainerServiceServicePrincipalProfileHash,
+	}
+
+	values := map[string]interface{}{}
+
+	values["client_id"] = profile.ClientID
+	values["client_secret"] = profile.Secret
+
+	servicePrincipalProfiles.Add(values)
+
+	return &servicePrincipalProfiles
+}
+
+func flattenAzureRmContainerServiceDiagnosticsProfile(profile *containerservice.DiagnosticsProfile) *schema.Set {
+	diagnosticProfiles := schema.Set{
+		F: resourceAzureRMContainerServiceDiagnosticProfilesHash,
+	}
+
+	values := map[string]interface{}{}
+
+	values["enabled"] = profile.VMDiagnostics.Enabled
+	values["storage_uri"] = profile.VMDiagnostics.StorageURI
+	diagnosticProfiles.Add(values)
+
+	return &diagnosticProfiles
+}
+
+func expandAzureRmContainerServiceDiagnostics(d *schema.ResourceData) (containerservice.DiagnosticsProfile, error) {
```

Does this need to return an error?